torch/csrc/jit/passes/memory_planning/memory_observer.h

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+#include <ATen/record_function.h>
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/runtime/interpreter.h>
+
+namespace torch {
+namespace jit {
+
+inline int64_t timeSinceEpoch() {
+  return std::chrono::duration_cast<std::chrono::microseconds>(
+             std::chrono::system_clock::now().time_since_epoch())
+      .count();
+}
+
+typedef struct TORCH_API MemoryEvent {
```

Comment:
Cool 😎 😎 😎 

Could you add some high-level comments to this file ?

Reply:
will do